### PR TITLE
Feature/transaction

### PR DESCRIPTION
**Pull request by eg8451**

### CODE REVIEW

File: src/main/java/com/e451/domain/Transaction.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    private Long quantity;
+    private Double salesValue;
+    private Long storeId;
+    private Double couponMatchDiscount;
+    private Double couponDisc;
+    private Double retailDisc;
+    private Long transTime;
+    private Long weekNo;
+
+    public Transaction(Long id, Long householdKey, Long basketId, Long day, Long productId, Long quantity, Double salesValue, Long storeId, Double couponMatchDiscount, Double couponDisc, Double retailDisc, Long transTime, Long weekNo) {
```
Comment (johnmuller-8451) [inline src/main/java/com/e451/domain/Transaction.java]: This is a really long line of code. Maybe break it up into multiple lines to make it more readable:

```
    public Transaction(Long id, Long householdKey, Long basketId,
                      Long day, Long productId, Long quantity,
                      ...) {
    ...
    }
```

File: src/main/java/com/e451/models/TransactionRecord.java [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    private Long storeId;
+    private Double couponMatchDiscount;
+    private Double couponDisc;
+    private Double retailDisc;
+    private Long transTime;
+    private Long weekNo;
+
+    public TransactionRecord() {}
+
+    public TransactionRecord(Long id, Long householdKey, Long basketId, Long day, Long productId, Long quantity, Double salesValue, Long storeId, Double couponMatchDiscount, Double couponDisc, Double retailDisc, Long transTime, Long weekNo) {
```
Comment (johnmuller-8451) [inline src/main/java/com/e451/models/TransactionRecord.java]: Another really long line of code.

File: src/main/java/com/e451/controller/TransactionController.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import org.springframework.data.web.PagedResourcesAssembler;
+import org.springframework.hateoas.PagedResources;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+/**
```
Comment (johnmuller-8451) [inline src/main/java/com/e451/controller/TransactionController.java]: The boilerplate comment that IntelliJ adds should be deleted before you commit your code.

File: src/test/java/com/e451/service/TransactionServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+                PRODUCT_ID,
+                QUANTITY,
+                SALES_VALUE,
+                STORE_ID,
+                COUPON_MATCH_DISCOUNT,
+                COUPON_DISC,
+                RETAIL_DISC,
+                TRANS_TIME,
+                WEEK_NO
+        ));
```
Comment (johnmuller-8451) [inline src/test/java/com/e451/service/TransactionServiceTest.java]: Please make your parens and whitespace usage consistent on method calls. In some places, you have your closing parens on the same line as the final parameter. In others, you have it on a separate line. (Incidentally, I'd prefer them on the same line as the final parameter, but as long as it's consistent across the file, I'm ambivalent.)

File: src/test/java/com/e451/service/TransactionServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+        Assert.assertEquals("The day was incorrectly converted", DAY, Transaction.getDay());
+        Assert.assertEquals("The productId was incorrectly converted", PRODUCT_ID, Transaction.getProductId());
+        Assert.assertEquals("The quantity was incorrectly converted", QUANTITY, Transaction.getQuantity());
+        Assert.assertEquals("The salesValue was incorrectly converted", SALES_VALUE, Transaction.getSalesValue(), 0);
+        Assert.assertEquals("The storeId was incorrectly converted", STORE_ID, Transaction.getStoreId());
+        Assert.assertEquals("The couponMatchDiscount was incorrectly converted", COUPON_MATCH_DISCOUNT, Transaction.getCouponMatchDiscount(), 0);
+        Assert.assertEquals("The retailDsic was incorrectly converted", RETAIL_DISC, Transaction.getRetailDisc(), 0);
+        Assert.assertEquals("The transTime was incorrectly converted", TRANS_TIME, Transaction.getTransTime());
+        Assert.assertEquals("The weekNo was incorrectly converted", WEEK_NO, Transaction.getWeekNo());
+
```
Comment (johnmuller-8451) [inline src/test/java/com/e451/service/TransactionServiceTest.java]: All this extra whitespace at the end of the method can be deleted.

File: src/test/java/com/e451/service/TransactionServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+    private final Long STORE_ID2 = new Long(6);
+    private final Double COUPON_MATCH_DISCOUNT2 = .25;
+    private final Double COUPON_DISC2 = .15;
+    private final Double RETAIL_DISC2 = .07;
+    private final Long TRANS_TIME2 = new Long(1725);
+    private final Long WEEK_NO2 = new Long(7);
+
+    private TransactionService testTransactionService;
+    private Pageable mockPageable = mock(Pageable.class);
+    private TransactionRepository TransactionRepository = mock(TransactionRepository.class);
```
Comment (johnmuller-8451) [inline src/test/java/com/e451/service/TransactionServiceTest.java]: Variable names should begin with a lowercase letter, and mocks should generally have the word "mock" in the variable name to make it obvious that they aren't real object instances. For example: `private TransactionRepository mockTransactionRepository = mock(TransactionRepository.class);`

Please make variable capitalization changes throughout the file.

**Review by johnmuller-8451 — DISMISSED** [review]
Mostly formatting changes, but things that nevertheless should be fixed to adhere to coding standards.

**Review by HLaiveling — APPROVED** [review]
Looks good

**Review by tomdantuit — APPROVED** [review]
Looks like all the changes were made.

**Pull request merged**